addons/misra.py

```diff
@@ -1656,6 +1659,148 @@ def get_violations(self):
+                    for each in ruleItemList:
+                        if each is not None:
+                            if (each[0] == line_symbol[0]) and (each[1] == line_symbol[1]):
+                                matched = True
+
+                    # Append the rule item if it was not already found
+                    if not matched:
+                        ruleItemList.append(line_symbol)
+
+
+    def isRuleSuppressed(self, location, ruleNum):
+        """
+        Check to see if a rule is suppressed.
+        ruleNum is the rule number in hundreds format
+
+        If the rule exists in the dict then check for a filename
+        If the filename is None then rule is suppressed globally
+        for all files.
+        If the filename exists then look for list of
+        line number, symbol name tuples.  If the list is None then
+        the rule is suppressed for the entire file
+        If the list of tuples exists then search the list looking for
+        maching line numbers.  Symbol names are currently ignored
+        because they can include regular expressions.
+        TODO: Support symbol names and expression matching.
+
+        """
+        ruleIsSuppressed = False
+        filename = location.file
+        linenr    = location.linenr
+
+        if ruleNum in self.suppressedRules:
+            fileDict = self.suppressedRules[ruleNum]
+
+            # a file name entry of None means that the rule is suppressed
+            # globally
+            if None in fileDict:
+                ruleIsSuppressed = True
+            else:
+                # Does the filename match one of the names in
+                # the file list
+                if filename in fileDict:
+                    # Get the list of ruleItems
+                    ruleItemList = fileDict[filename]
+
+                    if ruleItemList is None:
+                        # None for itemRuleList means the rule is suppressed
+                        # for all lines in the filename
+                        ruleIsSuppressed = True
+                    else:
+                        # Iterate though the the list of line numbers
+                        # and symbols looking for a match of the line
+                        # number.  Matching the symbol is a TODO:
+                        for each in ruleItemList:
+                            if each is not None:
+                                if each[0] == linenr:
+                                    ruleIsSuppressed = True
+
+        return ruleIsSuppressed
+
+
+    def parseSuppressions(self):
+        """
+        Parse the suppression list provided by cppcheck looking for
+        rules that start with 'misra' or MISRA.  The MISRA rule number
+        follows using either '_' or '.' to separate the numbers.
+        Examples:
+            misra_6.0
+            misra_7_0
+            misra.21.11
+        """
+        rule_pattern = re.compile(r'([a-zA-Z0-9]+)[_.]([0-9]+)[_.]([0-9]+)')
```

Comment:
it seems to me that the pattern should start with misra. something like:

    r'misra[_.]([0-9]+)[_.]([0-9]+)'